src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -377,6 +377,10 @@ static bool ReferencesNServiceBus(Assembly assembly, List<AssemblyName> processe
                 {
                     continue;
                 }
+                catch (FileLoadException)
```

Comment:
I cherry picked the changes from @distantcam and while it's not directly related it seems to be a good change. Maybe Cam can share some insights how he was able to get that exception?